src/lib/components/ServerSentEvents.react.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+/** An interface to server sent events in Dash */
+export default class ServerSentEvents extends React.Component {
+    componentDidMount() {
+        const { url, withCredentials, setProps } = this.props;
+
+        this.eventSource = new EventSource(url, { withCredentials });
+
+        this.eventSource.onmessage = (event) => {
+            setProps({ message: event.data });
```

Comment:
Is there a reason why you return the event data for messages, but a JSON object for errors?

Reply:
(I guess it would be most coherent to do one or the other)

Reply:
I was not sure which properties to use for error event. Then I looked at the WebSocket component and borrowed from there.